src/infi/clickhouse_orm/query.py

```diff
@@ -303,14 +365,11 @@ def order_by_as_sql(self):
             for field in self._order_by
         ])
 
-    def conditions_as_sql(self):
+    def conditions_as_sql(self, q_object):
```

Comment:
To prevent breaking the public API, I suggest changing the method signature to:
`def conditions_as_sql(self, prewhere=False)`
And return either the "where" SQL or the "prewhere" SQL based on the flag.

Reply:
I'm not sure about future `having` filters adding. But if going django way and add it into filter syntax (on already aggregated fields) it can work.

Reply:
To be true, I don't see any need in this method, if all conditions are Q objects with as_sql method inside.

Reply:
Another thing I'm not sure about, why this method has public area of visibility, not protected, as well as other methods that build a part of the query.